server/src-lib/Hasura/Server/Auth/JWT.hs

```diff
@@ -371,28 +463,103 @@ processAuthZHeader jwtCtx@JWTCtx{jcxClaimNs, jcxClaimsFormat} authzHeader = do
 
     malformedAuthzHeader =
       throw400 InvalidHeaders "Malformed Authorization header"
-    claimsNotFound = do
-      let claimsNsError = case jcxClaimNs of
-                            ClaimNsPath path -> T.pack $ "claims not found at claims_namespace_path: '"
-                                                <> encodeJSONPath path <> "'"
-                            ClaimNs ns -> "claims key: '" <> ns <> "' not found"
-      throw400 JWTInvalidClaims claimsNsError
 
-
--- parse x-hasura-allowed-roles, x-hasura-default-role from JWT claims
-parseHasuraClaims :: forall m. (MonadError QErr m) => J.Object -> m HasuraClaims
-parseHasuraClaims claimsMap = do
-  HasuraClaims <$>
-    parseClaim allowedRolesClaim "should be a list of roles" <*>
-    parseClaim defaultRoleClaim  "should be a single role name"
+-- ^ parse the claims map from the JWT token or custom claims from the JWT config
+parseClaimsObject
```

Comment:
The pytests are great! but I would really love to see hspec unit tests for this function. Without clear tests that define a spec for this function (or thorough documentation), I can't say whether it looks correct.

Testing this in the haskell tests would make it much easier for me to run locally, and look at code coverage as well.

Reply:
I have defined a spec for it, I have refactored the above function to `parseClaimsMap`.